### PR TITLE
ci: add golangci-lint workflow and fix lint errors

### LINKED ISSUE #29: CI: introduce  golangci-lint and gofmt

### PR DESCRIPTION
**Pull request by nottrif**
## Changes done:
- Added golangci-lint.yaml file which specifies the workflow.
- Added .gitattributes to ensure that line endings for windows builds are properly formatted.

## Tests done:
- Verified that the lint workflow is being triggered upon pull request. 
- Verified that annotations were created by the workflow upon errors encountered.

## Screenshots:
Screenshot_1339
Screenshot_1340
Screenshot_1341

### Note to reviewer: 
Any feedback would be very much appreciated.

closes #29

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@nottrif Thanks for the PR. Could you fix the lint issues in this PR itself?

**Comment by tao12345666333** [comment]
thanks

**Comment by nottrif** [comment]
Yes, sure.

**Comment by nottrif** [comment]
I wanted to validate if removing unused functions and fields from files is okay, the initial check found 4 unused functions, and so I decided to purge them and then made commits to my fork's branch, the workflow ran and found 2 unused fields in another file.

**Comment by nottrif** [comment]
Haha, I just realized that I should have integrated gofmt before beginning to fix lint errors.

File: pkg/api/apisix/client.go [hunk 1]
```diff
@@ -114,52 +114,6 @@ func readBody(r io.ReadCloser) (string, error) {
 	return string(data), nil
 }
 
-// getSchema returns the schema of APISIX object.
-func (c *Client) getSchema(ctx context.Context, url string) (string, error) {
```
Comment (lingsamuel) [inline pkg/api/apisix/client.go:118]: this two functions should be kept

File: test/cli/suites-consumer/sdk.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ var _ = ginkgo.Describe("adc APISIX consumer SDK tests", func() {
 			baseConsumer1 := &types.Consumer{
 				Username: "consumer1",
 			}
-			_, err = s.CreateConsumer(baseConsumer1)
```
Comment (lingsamuel) [inline test/cli/suites-consumer/sdk.go:33]: this error should be checked

File: test/cli/suites/dump.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ version: ""
 			gomega.Expect(err).To(gomega.BeNil(), "check route delete")
 			err = s.DeleteService("svc1")
 			gomega.Expect(err).To(gomega.BeNil(), "check service delete")
-			err = s.DeleteService("svc2")
```
Comment (lingsamuel) [inline test/cli/suites/dump.go:72]: this error should be checked

**Review by lingsamuel — COMMENTED** [review]
Thanks for your help, just some small comments.

**Comment by tao12345666333** [comment]
You can also ignore some default rules by adding golangci-lint configuration file

**Comment by nottrif** [comment]
Oh okay, I'll check the configurations.

File: pkg/api/apisix/client.go (hunk 1 above)
Comment (nottrif) [reply pkg/api/apisix/client.go:118]: those commits have been reverted and the functions have been reverted.

**Comment by tao12345666333** [comment]
Please resolve the conflicts, thanks!

**Comment by nottrif** [comment]
Hi. I was able to change the config to check for new issues only. :) Regarding gofmt, I changed my VSCode settings (`ctrl+,` -> `editor.formatOnSave`) and defaulted it to save go files by **gofmt** standards and all the files were already formatted the right way. I will resolve the error checks mentioned above and close this by today, thanks for your patience.

**Comment by tao12345666333** [comment]
Thanks :+1:

**Comment by tao12345666333** [comment]
Could you please sign the CLA again?

File: cmd/validate.go [hunk 4]
```diff
@@ -105,7 +105,7 @@ func validateContent(c *types.Configuration) error {
 		return err
 	}
 	errs := v.Validate()
-	if errs != nil && len(errs) > 0 {
+	if len(errs) > 0 {
```
Comment (tao12345666333) [inline cmd/validate.go:108]: we should keep the `errs != nil` condition

File: .github/workflows/golangci-lint.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
```
Comment (tao12345666333) [inline .github/workflows/golangci-lint.yml:5]: we don't need the master branch.

**Comment by nottrif** [comment]
I've signed it and resolved the comments you mentioned.

File: test/cli/suites-consumer/sdk.go (hunk 2 above)
Comment (nottrif) [reply test/cli/suites-consumer/sdk.go:33]: I have used gomega to assert if errors are nil and entailed an optionalDescription if they are not matching. can you please review the error checks?

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Pull request merged**